index.html

```diff
@@ -316,14 +316,22 @@ <h3>Normative Specifications</h3>
                     <dt id="integrity" class="spec">Linked Data Integrity (LDI)</dt>
                     <dd>
                         <p>
-                            This specification defines a generic framework for expressing proofs of integrity of Linked Datasets via, for example, digital signatures, proofs of work, or proofs of existence. Beyond the generic (normative) framework, this deliverable also provides a normative definition for Linked Data Signatures, as well as a means to express these proofs in an RDF Dataset in a way that enables a 3rd party to verify the integrity of the data.
+                            This specification defines a structure for embedding signatures in RDF documents.
+                            The group will define that embedding mechanism to work with RDH, though the hashing algorithm will be identified as an assertion in the structure, allowing the same structure to be used with other hashing algorithms.
+                            This deliverable enables 3rd parties to verify that the data has not changed since it was signed.
+                            Time-permitting, this signature may be designed to express proofs of work or proofs of existance.
```

Comment:
> I am no expert in crypto, but from the discussion we had while preparing this charter, my understanding was that "proof (of X)" is not a kind of signature, but rather the opposite...

That is correct. My understanding is that there are other proof algorithms that are _not_ based on the traditional form of signature. This deliverable will not _standardize_ any other proof mechanism than the one based on signatures, but the overall framework should be general enough so that, later, other mechanisms can be "plugged in".

I let those who understand this stuff better (@peacekeeper, @msporny, @dlongley) comment on the PR.

Thanks @pchampin for having drawn attention to this.

Reply:
My understanding is that traditional signatures are one type of proof, and that there are other types of proofs as well that don't involve signing (e.g. proofs-of-existence when publishing a hash of an RDF dataset on a distributed ledger).

In addition, there can also be types of proofs that don't hash the entire RDF dataset, but hash individual RDF statements or subsets of a RDF dataset (e.g. see https://w3c-ccg.github.io/ldp-bbs2020/#create-verify-data-algorithm).

The RDC, RDH, and LDI deliverables should probably be flexible enough to allow any of the above types of proofs. From this perspective, I'm not sure if the current PR sufficiently enables this flexibility?

Reply:
> I am no expert in crypto, but from the discussion we had while preparing this charter, my understanding was that "proof (of X)" is not a kind of signature, but rather the opposite...

I included the sentence with "proofs of work or proofs of existence" from the original text to avoid controversy. Danbri also points out that "proofs of work" runs afoul of a TAG decision not to melt the planet. Perhaps the best way forward is to refine this PR while keeping them and lobby for their removal (which I support) later.

Reply:
> The RDC, RDH, and LDI deliverables should probably be flexible enough to allow any of the above types of proofs. From this perspective, I'm not sure if the current PR sufficiently enables this flexibility?

My intention was that "embedding mechanism to work with RDH, though the hashing algorithm will be identified as an assertion in the structure, allowing the same structure to be used with other hashing algorithms" captured that. The current text leaves room for it with "generic framework for expressing proofs" but doesn't say that that LDI will support RDH (and a "framework" can be anything).

Can you think of a better way to say that LDI supports at least RDH, though can be used with other claims"?

Reply:
> Danbri also points out that "proofs of work" runs afoul of a TAG decision not to melt the planet.

Not all proofs of work melt the planet... in fact, for 20+ years, very large swaths of the Web have depended on proof of work to protect themselves against attack, and even today depend on proof of work to operate. Most website password checking systems use proof of work in the form of the bcrypt algorithm:

https://en.wikipedia.org/wiki/Bcrypt

This is still an active area of research and standardization:

https://en.wikipedia.org/wiki/Argon2

I'm certainly no Bitcoin/Ethereum maximalist -- I do believe those networks are actively hostile towards the health of our planet (and there are better alternatives once people come to their senses)... BUT, we shouldn't take a position against all proof of work algorithms; they have their place on the Web and have for decades.

Reply:
@msporny if you look at my proposal for this paragraph in https://github.com/w3c/lds-wg-charter/pull/73#pullrequestreview-657731438 it simply ducks this issue. There is no reason to raise flags here when a simple "for example" will do (proposed, actually, by @rhiaro on a separate mail...)

Reply:
> > Danbri also points out that "proofs of work" runs afoul of a TAG decision not to melt the planet.
> 
> Not all proofs of work melt the planet... in fact, for 20+ years, very large swaths of the Web have depended on proof of work to protect themselves against attack, and even today depend on proof of work to operate. Most website password checking systems use proof of work in the form of the bcrypt algorithm:
> 
> https://en.wikipedia.org/wiki/Bcrypt
> 
> This is still an active area of research and standardization:
> 
> https://en.wikipedia.org/wiki/Argon2
> 
> I'm certainly no Bitcoin/Ethereum maximalist -- I do believe those networks are actively hostile towards the health of our planet (and there are better alternatives once people come to their senses)... BUT, we shouldn't take a position against all proof of work algorithms; they have their place on the Web and have for decades.

Yeah just check in with the TAG to make sure the WG's expectations are clear, and clearly communicated to groups considering joining and voting for the group, and that planet-hostile forms of proof of work are out of scope. 

The easiest way to do that would be to cite their https://www.w3.org/2001/tag/doc/ethical-web-principles/#sustainable e.g. in the sentence mentioning proofs of work.